libs/shared/src/main/java/io/crate/common/Booleans.java

```diff
@@ -122,4 +125,28 @@ public static boolean isFalse(String value) {
     public static boolean isTrue(String value) {
         return "true".equals(value);
     }
+
+    /**
+     * Copies a collection of {@code Boolean} instances into a new array of primitive {@code boolean}
+     * values.
+     *
+     * <p>Elements are copied from the argument collection as if by {@code collection.toArray()}.
+     * Calling this method is as thread-safe as calling that method.
+     *
+     * <p><b>Note:</b> consider representing the collection as a {@link java.util.BitSet} instead.
+     *
+     * @param collection a collection of {@code Boolean} objects
+     * @return an array containing the same values as {@code collection}, in the same order, converted
+     *     to primitives
+     * @throws NullPointerException if {@code collection} or any of its elements is null
+     */
+    public static boolean[] toArray(Collection<Boolean> collection) {
+        Object[] boxedArray = collection.toArray();
+        int len = boxedArray.length;
+        boolean[] array = new boolean[len];
+        for (int i = 0; i < len; i++) {
+            array[i] = (Boolean) Objects.requireNonNull(boxedArray[i]);
+        }
+        return array;
+    }
```

Comment:
```suggestion
    public static boolean[] toArray(List<Boolean> collection) {
        int len = collection.size();
        boolean[] array = new boolean[len];
        for (int i = 0; i < len; i++) {
            array[i] = (Boolean) Objects.requireNonNull(collection.get(i));
        }
        return array;
    }
```

Looks like all call sites have a `List`. Allows us to avoid the `toArray` copy